### PR TITLE
[#7] Clean osolete environments

### PR DESCRIPTION
**Pull request by siniann**
## Type of request

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring 
- [ ] Documentation or documentation changes

#### Description of the change
Clean obsolete worktrees 

#### Description of current state
Functionlity to Clean obsolete worktrees is not available
#### Description of target state
Add a menu item to provide the clean up functionality. Allow the user to choose the obsolete worktrees to delete.
#### Initiator
name / company / position / created at

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

File: bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/dialog/progress/internal/ObsoleteEnvironmentDialog.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        checkboxTreeViewer.expandAll();
+        TreeColumn[] columns = checkboxTreeViewer.getTree().getColumns();
+        for (TreeColumn column : columns) {
+            column.pack();
+        }
+        checkboxTreeViewer.refresh();
+        return tableComposite;
+    }
+
+
```
Comment (d-gs) [inline bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/dialog/progress/internal/ObsoleteEnvironmentDialog.java:158]: Nitpicking comment: extra new-line.

File: bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/message/messages.properties [hunk 2]
```diff
@@ -98,3 +98,14 @@ featureBranchWorkflowView.label.dashboardViewTargetBranchColumn = Target Branch
...
+# CleanObsoleteEnvironments
+cleanObsoleteEnvironmentsHandler.message.fetchWorktreesProgressMonitor = Fetching obsolete work trees
+
+cleanObsoleteEnvironmentsHandler.title.commandProgressDialog = Remove Obsolete Worktrees
+cleanObsoleteEnvironmentsHandler.message.noWorktreesAvailableDialog = There are no obsolete work trees to remove!
+cleanObsoleteEnvironmentsHandler.message.confirmDeleteDialog = Do you want to delete the selected worktrees?
+cleanObsoleteEnvironmentsHandler.message.nothingSelectedDialog = Nothing was selected to delete.
+
+worktree.label = Worktree Location
+branch.label = Branch
```
[the remark this answers is not in the document]
Comment (d-gs) [reply bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/message/messages.properties:111]: Actually the branch one is not necessary if the column is removed.

**Review by d-gs — CHANGES_REQUESTED** [review]
Thanks a lot for your work. It looks like we can refine the workflow for the end-user a bit, avoiding to inform when nothing is selected by:

* Do not allow to cancel the fetch of the obsolete environments
* Disabling the OK button and not doing anything while closing and/or cancel on the dialog
* Inform the user if some worktree launcher cannot be launch (meaning that the obsolete environment cannot be removed)
* Showing before start the deletion the list of environments to be removed

The other comments are to cleanup, reduce the information for the end-user about internal concepts and i18n improvements.

Back to you @siniann!

File: bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/handler/CleanObsoleteEnvironmentsHandler.java [hunk 3]
```diff
@@ -61,42 +71,41 @@ private void execute(final Session session, final Shell shell) {
...
-                        if (!cmds.isEmpty()) {
-                            CommandProgressDialog.open(shell,
-                                    messages.cleanObsoleteEnvironmentsHandler_title_commandProgressDialog,
-                                    messages, cmds);
+                final String selectionString = selectedItems.stream() //
+                        .map(env -> "\t- " + env.getDisplayName()) //
+                        .collect(Collectors.joining("\n"));
+                boolean openConfirm = MessageDialog.openConfirm(shell,
+                        bundleProperties.menuCleanObsoleteenvironments_label,
+                        MessageFormat.format(messages.cleanObsoleteEnvironmentsHandler_message_confirmDeleteDialog,
```
Comment (d-gs) [inline bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/handler/CleanObsoleteEnvironmentsHandler.java:92]: If it is formatted, the second part of the variable should be `messageFormat`.

File: bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/message/messages.properties [hunk 4]
```diff
@@ -98,3 +98,14 @@ featureBranchWorkflowView.label.dashboardViewTargetBranchColumn = Target Branch
 featureBranchWorkflowView.tooltip.dashboardViewTargetBranchColumn = Target feature branch for the selection
 featureBranchWorkflowView.label.dashboardViewWorktreeDefinitionColumn = Worktree definition
 featureBranchWorkflowView.tooltip.dashboardViewWorktreeDefinitionColumn = Repository name and branch at the worktree
+
+# CleanObsoleteEnvironments
+cleanObsoleteEnvironmentsHandler.message.fetchWorktreesProgressMonitor = Fetching obsolete work trees
```
[the remark this answers is not in the document]
Comment (d-gs) [reply bundles/org.aposin.gem.ui/src/org/aposin/gem/ui/message/messages.properties:104]: Not fixed yet.

**Review by d-gs — COMMENTED** [review]
I leave some comments for tomorrow. @davu-all will take over the review

File: bundles/org.aposin.gem.ui/OSGI-INF/l10n/bundle.properties [hunk 5]
```diff
@@ -14,3 +14,5 @@ menu.file.quit.label = Quit
 menu.help.label = Help
 menu.help.about.label = About
 menu.theme.label = Theme
+menu.clean.label = Clean
+menu.clean.obsoleteenvironments.label = Obsolete Environments
```
[the remark this answers is not in the document]
Comment (davu-all) [reply bundles/org.aposin.gem.ui/OSGI-INF/l10n/bundle.properties:18]: Done

**Review by davu-all — APPROVED** [review]
LGTM

**Pull request merged**